src/query/ast/src/parser/query.rs

```diff
@@ -316,6 +316,10 @@ impl<'a, I: Iterator<Item = WithSpan<'a, SetOperationElement>>> PrattParser<I>
                 if query.offset.is_some() {
                     return Err("ORDER BY must appear before OFFSET");
                 }
+                let order_by = order_by
```

Comment:
we need filter out `Expr::Constant`

Reply:
Do not modify the parser, this is a semantic problem. It's better to deal with in the binder phase, check whether the `rewrite_scalar` is `ScalarExpr::ConstantExpr`.
https://github.com/databendlabs/databend/blob/main/src/query/sql/src/planner/binder/sort.rs#L150